### DISCUSSION TITLE
03. PyTorch Computer Vision . Prepeare a datamodel problem

### DISCUSSION THREAD

**Discussion by genprokuror** [mrdbourke/pytorch-deep-learning#811] [Q&A]
Hey i just reviewed one part of code. 
`Check out what's inside the training dataloader
train_features_batch, train_labels_batch = next(iter(train_dataloader))
train_features_batch.shape, train_labels_batch.shape`
I cant understand why it gives this when launching
`(torch.Size([32, 1, 28, 28]), torch.Size([32]))`
Why theyr sizes and values are not the same? Dont author assigned them together here? How it works? 
`train_features_batch, train_labels_batch = next(iter(train_dataloader))`

**Comment by ZephyrusBlaze** [comment]
In PyTorch, the training dataloader typically yields batches of input features and their corresponding labels. In your case:

- `train_features_batch` has a shape of `(32, 1, 28, 28)`, indicating it's a batch of 32 samples, each with 1 channel (grayscale image) and a size of 28x28 pixels.
- `train_labels_batch` has a shape of `(32,)`, indicating it's a batch of 32 labels.

The sizes aren't the same because each element in `train_features_batch` corresponds to an image, while each element in `train_labels_batch` corresponds to the label of the respective image.

When you iterate over `train_dataloader`, it returns batches of features and labels, which you then unpack into `train_features_batch` and `train_labels_batch`. So, yes, they are assigned together, but they represent different aspects of your dataset: features and labels.

**Comment by LuluW8071** [comment]
To put it simply,
`Train_dataloader` contains tensor images  `(color_channels, height, width)` 
After, 
```python
 train_features_batch, train_labels_batch = next(iter(train_dataloader))
```

`Train_features_batch` contains images tensor arrays accompanied by batch size
`(Batch_size, color_channels, height, width)`

While, `Train_labels_batch` contain labels of those image data
 - `index of image labels` accompanied by batch size
`(Batch_size)`

The code snippet is just to assign batch_size (32) to load 32 images and respective 32 labels at a time in training device for model training.
